Modules/Common/src/ReferenceComparatorTask.cxx

```diff
@@ -0,0 +1,523 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    ReferenceComparatorTask.xx
+/// \author  Andrea Ferrero
+/// \brief   Post-processing task that compares a given set of plots with reference ones
+///
+
+#include "Common/ReferenceComparatorTask.h"
+#include "Common/TH1Ratio.h"
+#include "Common/TH2Ratio.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/DatabaseInterface.h"
+#include "QualityControl/ActivityHelpers.h"
+// ROOT
+#include <TClass.h>
+#include <TH1.h>
+#include <TH2.h>
+#include <TCanvas.h>
+
+using namespace o2::quality_control::postprocessing;
+using namespace o2::quality_control::core;
+using namespace o2::quality_control;
+
+namespace o2::quality_control_modules::common
+{
+
+static bool splitObjectPath(std::string fullPath, std::string& path, std::string& name)
+{
+  std::string delimiter = "/";
+  std::string det;
+  size_t pos = fullPath.rfind(delimiter);
+  if (pos == std::string::npos) {
+    return false;
+  }
+  path = fullPath.substr(0, pos);
+  name = fullPath.substr(pos + 1);
+  return true;
+}
+
+//_________________________________________________________________________________________
+//
+// Helper function for retrieving the last MonitorObject for a give run number
+
+static std::shared_ptr<MonitorObject> getMOFromRun(repository::DatabaseInterface* qcdb, std::string fullPath, uint32_t run, Activity activity)
+{
+  uint64_t timeStamp = 0;
+  activity.mId = run;
+  const auto filterMetadata = activity_helpers::asDatabaseMetadata(activity, false);
+  const auto objectValidity = qcdb->getLatestObjectValidity(activity.mProvenance + "/" + fullPath, filterMetadata);
```

Comment:
@Barthelemy indeed! The call in the `getMOFromRun()` function only happens once at the start of the run, when the reference plot is searched and retrieved.
However, the code contains a similar call in `getMO()`, which is invoked at each trigger. Currently I could not find a more efficient way of retrieving the last valid object and checking its validity range in one single CCDB access, but this might well be due to my limited knowledge of the CCDB interface.
The issue here is that we cannot rely on the time stamp of the trigger to lookup the CCDB objects, in the case we use a timer-based trigger, since the validity of good objects might have expired shortly before the trigger is invoked.